crates/ruff_linter/src/rules/airflow/rules/removal_in_3.rs

```diff
@@ -646,6 +649,63 @@ fn removed_name(checker: &mut Checker, expr: &Expr, ranged: impl Ranged) {
     }
 }
 
+fn extract_name_from_slice(slice: &Expr) -> Option<String> {
+    match slice {
+        Expr::StringLiteral(ExprStringLiteral { value, .. }) => Some(value.to_string()),
+        _ => None,
+    }
+}
+
+pub(crate) fn removed_context_variable(checker: &mut Checker, expr: &Expr) {
+    const REMOVED_CONTEXT_KEYS: [&str; 12] = [
+        "conf",
+        "execution_date",
+        "next_ds",
+        "next_ds_nodash",
+        "next_execution_date",
+        "prev_ds",
+        "prev_ds_nodash",
+        "prev_execution_date",
+        "prev_execution_date_success",
+        "tomorrow_ds",
+        "yesterday_ds",
+        "yesterday_ds_nodash",
+    ];
+
+    if let Expr::Subscript(ExprSubscript { value, slice, .. }) = expr {
+        if let Expr::Name(ExprName { id, .. }) = &**value {
+            if id.as_str() == "context" {
```

Comment:
I think we'd also need to check where the `context` variable is originating from otherwise, I think, this will raise a violation on all variables that's named "context" and is using a similar access pattern.

Reply:
i.e., we need to make sure that the definition of `context` variable is the function parameter that's decorated with `@task`.